Python/gc_free_threading.c

```diff
@@ -1952,25 +1953,23 @@ int
 PyGC_Enable(void)
 {
     GCState *gcstate = get_gc_state();
-    int old_state = gcstate->enabled;
-    gcstate->enabled = 1;
+    int old_state = _Py_atomic_exchange_int(&gcstate->enabled, 1);
```

Comment:
You can remove the old_state variable. Same remark in PyGC_Disable() and PyGC_IsEnabled().